packages/concerto-core/lib/decoratorextractor.js

```diff
@@ -24,21 +24,49 @@ const { MetaModelNamespace } = require('@accordproject/concerto-metamodel');
  * @memberof module:concerto-core
  */
 class DecoratorExtractor {
+    /**
+     * The action to be performed to extract all, only vocab or only non-vocab decorators
+     */
+    static Action = {
```

Comment:
Can you please share your motivation behind this? What problem/blocker does this aim to solve?

Reply:
Here the idea is to reduce code duplication by using some flags.
Initially I thought of having separate flags for each operation but it made more sense to use just a single flag instead of 3 as only one of them will be set at a time.

Reply:
I think we can have an `options` object as argument instead. That we can stick to existing pattern.
```